### PR TITLE
Enable preview features

### LINKED ISSUE #9: --enable-preview through Jabel?
When I use `--enable-preview` I cannot start the compilation with Jabel. It complains that Java 13 preview features are not possible to use with Java 8 release.

Could Jabel have an option so that it will also tell the compiler to handle the preview features and still generate Java 8 binary?

### PR DESCRIPTION
**Pull request by astubbs**
Fixes #9 --enable-preview through Jabel?
First cut, works for me so far...

~~Branched from logging PR, can rebase on master if you like instead: https://github.com/bsideup/jabel/pull/37~~

### CODE REVIEW

File: jabel-javac-plugin/src/main/java/com/github/bsideup/jabel/PreviewFeatureCheckOverride.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package com.github.bsideup.jabel;
+
+import com.sun.tools.javac.code.Preview;
+import com.sun.tools.javac.code.Source;
+import net.bytebuddy.asm.Advice;
+
+public class PreviewFeatureCheckOverride {
```
Comment (bsideup) [inline jabel-javac-plugin/src/main/java/com/github/bsideup/jabel/PreviewFeatureCheckOverride.java:7]: Let's follow the convention of `${ClassBeingVisited}Advice`:

```suggestion
public class PreviewAdvice {
```

**Comment by bsideup** [comment]
Hey @astubbs,

Since the logging one may take longer, could you please rebase this PR on top of master? Thanks!

**Comment by astubbs** [comment]
Will do @bsideup , what do you think otherwise?

**Comment by astubbs** [comment]
Wow ok have been pulling my hair out about this. Turns out the difference of fail vs not failing (it passes locally no matter what I do, and fails on CI), was Gradle wrapper vs not. I /had/ java 14 installed on my machine and removing that made no difference.

I had a quick look through the release notes between 6.2 and 6.6 and couldn't find anything obvious...

```

↪ gradle -version

------------------------------------------------------------
Gradle 6.6
------------------------------------------------------------

Build time:   2020-08-10 22:06:19 UTC
Revision:     d119144684a0c301aea027b79857815659e431b9

Kotlin:       1.3.72
Groovy:       2.5.12
Ant:          Apache Ant(TM) version 1.10.8 compiled on May 10 2020
JVM:          13.0.2 (Oracle Corporation 13.0.2+8)
OS:           Mac OS X 10.15.6 x86_64

11:55:48 18/08/20 in /Users/antony/repos/projects/jabel
↪ gradle clean test

Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/6.6/userguide/command_line_interface.html#sec:command_line_warnings

BUILD SUCCESSFUL in 1s
7 actionable tasks: 6 executed, 1 up-to-date
11:55:53 18/08/20 in /Users/antony/repos/projects/jabel
↪ ./gradlew -version

------------------------------------------------------------
Gradle 6.2.1
------------------------------------------------------------

Build time:   2020-02-24 20:24:10 UTC
Revision:     aacbcb7e587faa6a8e7851751a76183b6187b164

Kotlin:       1.3.61
Groovy:       2.5.8
Ant:          Apache Ant(TM) version 1.10.7 compiled on September 1 2019
JVM:          13.0.2 (Oracle Corporation 13.0.2+8)
OS:           Mac OS X 10.15.6 x86_64

11:56:01 18/08/20 in /Users/antony/repos/projects/jabel
↪ ./gradlew clean test

> Task :example:compileJava FAILED
/Users/antony/repos/projects/jabel/example/src/main/java/com/example/JabelExample.java:16: error: switch expressions are a preview feature and are disabled by default.
        var result = switch (args.length) {
                     ^
  (use --enable-preview to enable switch expressions)
/Users/antony/repos/projects/jabel/example/src/main/java/com/example/JabelExample.java:17: error: switch rules are a preview feature and are disabled by default.
            case 1 -> {
                   ^
  (use --enable-preview to enable switch rules)
/Users/antony/repos/projects/jabel/example/src/main/java/com/example/JabelExample.java:20: error: multiple case labels are a preview feature and are disabled by default.
            case 2, 3 -> "two or three";
                  ^
  (use --enable-preview to enable multiple case labels)
/Users/antony/repos/projects/jabel/example/src/main/java/com/example/JabelExample.java:16: warning: as of release 10, 'var' is a restricted type name and cannot be used for type declarations or as the element type of an array
        var result = switch (args.length) {
            ^
/Users/antony/repos/projects/jabel/example/src/main/java/com/example/JabelExample.java:36: warning: as of release 10, 'var' is a restricted type name and cannot be used for type declarations or as the element type of an array
                Function<String, String> function = (var prefix) -> {
                                                     ^
/Users/antony/repos/projects/jabel/example/src/main/java/com/example/JabelExample.java:31: warning: as of release 10, 'var' is a restricted type name and cannot be used for type declarations or as the element type of an array
        var callable = new Callable<>() {
            ^
/Users/antony/repos/projects/jabel/example/src/main/java/com/example/JabelExample.java:44: warning: as of release 10, 'var' is a restricted type name and cannot be used for type declarations or as the element type of an array
        var closeable = new AutoCloseable() {
            ^
/Users/antony/repos/projects/jabel/example/src/main/java/com/example/JabelExample.java:54: error: variables in try-with-resources are not supported in -source 8
        try (closeable) {
                      ^
  (use -source 9 or higher to enable variables in try-with-resources)
/Users/antony/repos/projects/jabel/example/src/main/java/com/example/JabelExample.java:69: warning: as of release 10, 'var' is a restricted type name and cannot be used for type declarations or as the element type of an array
        var me = 4;
            ^
/Users/antony/repos/projects/jabel/example/src/main/java/com/example/JabelExample.java:70: warning: as of release 10, 'var' is a restricted type name and cannot be used for type declarations or as the element type of an array
        var operation = "squareMe";
            ^
/Users/antony/repos/projects/jabel/example/src/main/java/com/example/JabelExample.java:71: warning: as of release 10, 'var' is a restricted type name and cannot be used for type declarations or as the element type of an array
        var result = switch (operation) {
            ^
4 errors
7 warnings

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':example:compileJava'.
> Compilation failed; see the compiler error output for details.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/6.2.1/userguide/command_line_interface.html#sec:command_line_warnings

BUILD FAILED in 532ms
3 actionable tasks: 3 executed
11:56:07 18/08/20 in /Users/antony/repos/projects/jabel
↪
```

**Comment by astubbs** [comment]
Ok, I'm not actually sure this is needed to access preview features. I assumed from the existence of the issue, that it didn't so I dove straight in. Now I'm not so sure. I can't get it to test reliably...

**Comment by bsideup** [comment]
Hi @astubbs,

Please see my answer here:
https://github.com/bsideup/jabel/issues/9#issuecomment-694188484

**Comment by astubbs** [comment]
Haha ok so this definitely isn't needed - let's close it :) Was a fun learning experience though!

**Pull request closed without merging**